src/CoreMedia/CMFormatDescription.cs

```diff
@@ -374,7 +490,153 @@ internal CMAudioFormatDescription (IntPtr handle, bool owns)
 		{
 		}
 
-		// TODO: Move more audio specific methods here
+#if !COREBUILD
+		[DllImport (Constants.CoreMediaLibrary)]
+		static unsafe extern /* OSStatus */ CMFormatDescriptionError CMAudioFormatDescriptionCreate (
+			/* CFAllocatorRef */ IntPtr allocator, 
+			ref AudioStreamBasicDescription description,
+			nuint layoutSize, 
+			/* AudioChannelLayout* */ IntPtr layout, 
+			nuint magicCookieSize, 
+			/* void* */ IntPtr magicCookie, 
+			/* CFDictionaryRef* */ IntPtr extensions,
+			out IntPtr outDesc);
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd) : this (ref asbd, null, 0, IntPtr.Zero, null)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, AudioChannelLayout layout) : this (ref asbd, layout, 0, IntPtr.Zero, null)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, uint magicCookieSize, IntPtr magicCookie) : this (ref asbd, null, magicCookieSize, magicCookie, null)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, NSDictionary extensions) : this (ref asbd, null, 0, IntPtr.Zero, extensions)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, AudioChannelLayout layout, uint magicCookieSize, IntPtr magicCookie) : this (ref asbd, layout, magicCookieSize, magicCookie, null)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, AudioChannelLayout layout, NSDictionary extensions) : this (ref asbd, layout, 0, IntPtr.Zero, extensions)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, uint magicCookieSize, IntPtr magicCookie, NSDictionary extensions) : this (ref asbd, null, magicCookieSize, magicCookie, extensions)
+		{
+		}
+
+		public CMAudioFormatDescription (ref AudioStreamBasicDescription asbd, AudioChannelLayout layout, uint magicCookieSize, IntPtr magicCookie, NSDictionary extensions)
+		{
+			IntPtr ptr = IntPtr.Zero;
+			int size = 0;
+			if (layout != null) {
+				size = Marshal.SizeOf (layout.NativeStruct);
+				ptr = Marshal.AllocHGlobal (size);
```

Comment:
Who is deallocing this later?

Reply:
Can we not just pass the structure by ref instead of manually allocating marshal memory?